Modules/_sqlite/connection.c

```diff
@@ -1817,6 +1817,116 @@ pysqlite_connection_create_collation_impl(pysqlite_Connection *self,
     Py_RETURN_NONE;
 }
 
+#ifdef PY_SQLITE_HAVE_SERIALIZE
+/*[clinic input]
+_sqlite3.Connection.serialize as serialize
+
+    *
+    name: str = "main"
+        Which database to serialize.
+
+Serialize a database into a byte string. Non-standard.
+
+For an ordinary on-disk database file, the serialization is just a copy of the
+disk file. For an in-memory database or a "temp" database, the serialization is
+the same sequence of bytes which would be written to disk if that database
+were backed up to disk.
+[clinic start generated code]*/
+
+static PyObject *
+serialize_impl(pysqlite_Connection *self, const char *name)
+/*[clinic end generated code: output=97342b0e55239dd3 input=bb5513432e17fef8]*/
+{
+    if (!pysqlite_check_thread(self) || !pysqlite_check_connection(self)) {
+        return NULL;
+    }
+
+    sqlite3_int64 size;
+    const unsigned int flags = 0;
+    const char *data;
+
+    Py_BEGIN_ALLOW_THREADS
+    data = (const char *)sqlite3_serialize(self->db, name, &size, flags);
```

Comment:
reference: https://www.sqlite.org/c3ref/serialize.html

It sounds like the `SQLITE_SERIALIZE_NOCOPY` could provide a useful optimization, since we end up immediately copying the memory into a new bytes object. However, it would be more complex to use; most likely we'd have to first try with `SQLITE_SERIALIZE_NOCOPY` on, then if that fails, try again with it off. I don't know if that's worth the complexity, and in any case we're free to change it later if a user complains about the speed of repeatedly serializing a database.

Reply:
Let's try!

Reply:
See 7edccfed418eca5cdf88517cf9f3e91c2a051bc8.